cmd/bisync/bisync_test.go

```diff
@@ -275,7 +275,7 @@ func testBisync(t *testing.T, path1, path2 string) {
 
 	baseDir, err := os.Getwd()
 	require.NoError(t, err, "get current directory")
-	randName := "bisync-" + time.Now().Format("150405-") + random.String(5) // AzureBlob doesn't like dots
+	randName := "bs-" + time.Now().Format("150405-") + random.String(2) // some bucket backends don't like dots, keep this short to avoid linux errors
```

Comment:
This kind of implies that you are creating top level directories (which are buckets in s3 compatibles) - these have many limits under s3 compatibles.

I don't think you should be creating top level directories though, you should be doing this inside ones created by `fstest.RandomRemote` or using `fstest.RandomRemote` to create them.

Reply:
This is inside `fstest.RandomRemote` (for every remote except `local`.) For example, a path might end up looking like:

`TestS3:rclone-test-qemaser8watigiw0haguhij9/bisync-013220-biqaq/normalization/path1/file1.txt`

The listing filename ends up concatenating both paths, like:
`TestS3_rclone-test-qemaser8watigiw0haguhij9_bisync-013220-biqaq_extended_char_paths_path1_測試_Русский_____ě_áñ..TestS3_rclone-test-qemaser8watigiw0haguhij9_bisync-013220-biqaq_extended_char_paths_path2_測試_Русский_____ě_áñ.path1.lst`

which is why when we added `fstest.RandomRemote` a few days ago, it started causing `file name too long` errors locally (which I think is the underlying reason for the json decoding error which caused the endless loop.)

So basically I'm trying to make the path shorter by shortening the `bisync-013220-biqaq` part, to stay under that limit (and check errors properly so that worst case scenario it fails immediately.)

An alternative would be to use fewer than 24 random chars in `fstest.RandomRemote`...but I figured you wouldn't want to mess with that, and I don't think we have to.

Reply:
OK that is fine - I just wanted to make sure you weren't creating buckets :-)